### PR TITLE
ARROW-7014: [Developer][Release] Add "wheels" verification option to verify-release-candidate.sh for Linux and macOS

### PR DESCRIPTION
**Pull request by wesm**
Creates temporary conda environments to install and check the wheels. Checking within virtualenv would be more work but give us a stronger guarantee that things are working, but this will help catch obvious brokenness / corruption

My bash fu is not strong enough to conditionally check whether Flight or Gandiva are working on Python 3 wheels, maybe someone can help me.

For 0.15.1 rc0 this passes on Linux but fails on macOS because of the issue I cited in the vote thread.

### CODE REVIEW

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -99,14 +100,7 @@ fetch_archive() {
...
 
-test_binary() {
-  local download_dir=binaries
-  mkdir -p ${download_dir}
-
-  python3 $SOURCE_DIR/download_rc_binaries.py $VERSION $RC_NUMBER --dest=${download_dir}
-
-  pushd ${download_dir}
-
+verify_cwd_artifact_signatures() {
```
Comment (fsaintjacques) [inline dev/release/verify-release-candidate.sh]: The refactor make this function side effect free. Does it need to be cwd, can it just take the path as input?

File: dev/release/verify-release-candidate.sh [hunk 2]
```diff
@@ -531,6 +534,77 @@ test_binary_distribution() {
...
+    conda create -yq -n ${env} python=${py_arch//m/}
+    conda activate ${env}
+
+    pip install python-rc/${VERSION}-rc${RC_NUMBER}/pyarrow-${VERSION}-cp${py_arch//[m.]/}-cp${py_arch//./}-macosx_10_6_intel.whl
+
+    python -c "import pyarrow.parquet"
+    python -c "import pyarrow.plasma"
+    python -c "import pyarrow.fs"
+
+    # TODO check Flight only for Python 3
```
Comment (fsaintjacques) [inline dev/release/verify-release-candidate.sh]: Ditto with deactivate.

File: dev/release/verify-release-candidate.sh [hunk 3]
```diff
@@ -531,6 +534,77 @@ test_binary_distribution() {
...
+    python -c "import pyarrow.plasma"
+    python -c "import pyarrow.fs"
+
+    pip install python-rc/${VERSION}-rc${RC_NUMBER}/pyarrow-${VERSION}-cp${py_arch//[mu.]/}-cp${py_arch//./}-manylinux2010_x86_64.whl
+
+    python -c "import pyarrow.parquet"
+    python -c "import pyarrow.plasma"
+    python -c "import pyarrow.fs"
+
+    # TODO check Flight only for Python 3
```
Comment (fsaintjacques) [inline dev/release/verify-release-candidate.sh]:
```
$ for ver in 2.7 3.6m 3.7m; do if [[ "$ver" =~ ^3 ]]; then echo $ver; fi; done
3.6m
3.7m
```

**Review by fsaintjacques — APPROVED** [review]

File: dev/release/verify-release-candidate.sh (hunk 3 above)
Comment (wesm) [reply dev/release/verify-release-candidate.sh]: done

File: dev/release/verify-release-candidate.sh (hunk 2 above)
Comment (wesm) [reply dev/release/verify-release-candidate.sh]: done

File: dev/release/verify-release-candidate.sh (hunk 1 above)
Comment (wesm) [reply dev/release/verify-release-candidate.sh]: done

File: dev/release/verify-release-candidate.sh [hunk 4]
```diff
@@ -531,6 +534,77 @@ test_binary_distribution() {
...
+    python -c "import pyarrow.parquet"
+    python -c "import pyarrow.plasma"
+    python -c "import pyarrow.fs"
+
+    pip install python-rc/${VERSION}-rc${RC_NUMBER}/pyarrow-${VERSION}-cp${py_arch//[mu.]/}-cp${py_arch//./}-manylinux2010_x86_64.whl
+
+    python -c "import pyarrow.parquet"
+    python -c "import pyarrow.plasma"
+    python -c "import pyarrow.fs"
+
```
[the remark this answers is not in the document]
Comment (wesm) [reply dev/release/verify-release-candidate.sh]: done

**Comment by wesm** [comment]
Thanks for the bash tips! Merging

**Review by wesm — COMMENTED** [review]
.

**Pull request closed without merging**